mlflow/store/tracking/rest_store.py

```diff
@@ -197,28 +195,53 @@ def log_metric(self, run_id, metric):
         )
         self._call_endpoint(LogMetric, req_body)
 
+    def log_metric_async(self, run_id, metric):
```

Comment:
same feedback - not sure it's agreed that rest store entirely will take these APIs (i'm fine if they are).

Instead of this, an approach you can consider is to have the *client* try-call .log_async() and catch to call via queue.

That way, AML store can implement the method and your plugin will do what you want, everyone else is untouched/unmodified (from a contract perspective)